app/controllers/hq/partners_controller.rb

```diff
@@ -5,4 +5,47 @@ class Hq::PartnersController < ApplicationController
   def index
     @partners= Partner.all
   end
+
+  def edit
+    load_partner
+    load_associations
+  end
+
+  def update
+    load_partner
+    build_partner
+    save_partner or render 'edit'
+  end
+
+  private
+  def load_partner
+    @partner ||= Partner.find(params[:id])
+  end
```

Comment:
I was thinking:

def load_partner
  @partner ||= Partner.find_by_id(params[:id])
end

to return a partner or nil and then use it like you did for save:

load_partner or redirect_to(partners_path  , notice: "The specified partner was not found")

This way it does not leave the app. An it is useful for any future situations.